packages/components/src/Combobox/index.tsx

```diff
@@ -51,13 +51,30 @@ const Combobox = React.forwardRef(function ComboboxInner<T>(props: ComboboxProps
     inputElement,
     autoFocus = false,
     variant = 'outline',
+    minimumCharacters = 0,
     ...rest
   } = props;
   const [typedInputValue, setTypedInputValue] = useState(valueProp.toString());
   const { supported: voiceSupported } = useVoiceInput();
   const [value, setValue] = useState(valueProp.toString());
   const inputRef = useRef<HTMLInputElement | null>();
   const inAttachMode = Boolean(inputElement?.current);
+  const [items, setItems] = useState(itemsProp);
+  const lastItems = useRef(items);
+
+  useEffect(() => {
+    // Prevent flash of old dropdown items
+    if (value.length >= minimumCharacters && arraysEqual(items, []) && !arraysEqual(lastItems.current, itemsProp)) {
+      setItems(itemsProp);
+    }
+    lastItems.current = itemsProp;
+  }, [itemsProp]);
```

Comment:
Should we also include `minimumCharacters` in the array dependencies?

```suggestion
  }, [itemsProp, minimumCharacters]);
```